IonDotnet.Tests/Internals/TreeReaderTest.cs

```diff
@@ -0,0 +1,268 @@
+            //       90
+            //     ]
+            //   }
+            // }
+
+            var popupList = _ionValueFactory.NewEmptyList();
+            popupList.Add(_ionValueFactory.NewString("Open"));
+            popupList.Add(_ionValueFactory.NewString("Load"));
+            popupList.Add(_ionValueFactory.NewString("Close"));
+
+            var positionList = _ionValueFactory.NewEmptyList();
+            positionList.Add(_ionValueFactory.NewInt(1234));
+            positionList.Add(_ionValueFactory.NewInt(5678));
+            positionList.Add(_ionValueFactory.NewInt(90));
+
+            var deep3 = new IonStruct {{ "deep4val", _ionValueFactory.NewString("enddeep") }};
+
+            var deep2 = new IonStruct {{ "deep3", deep3 }};
+
+            var deep1 = new IonStruct {{ "deep2", deep2 }};
+
+            var menu = new IonStruct
+            {
+                { "id", _ionValueFactory.NewString("file") },
+                { "popup", popupList },
+                { "deep1", deep1 },
+                { "positions", positionList }
+            };
+
+            var value = new IonStruct {{ "menu", menu }};
+            var reader = new UserTreeReader(value);
+
+            ReaderTestCommon.Combined1(reader);
+        }
+
+        [TestMethod]
+        public void ValueWithAnnotationTest()
+        {
+            //Must be: {withannot: years::months::days::hours::minutes::seconds::18}
+            var intValue = _ionValueFactory.NewInt(18);
+            intValue.AddTypeAnnotation("years");
+            intValue.AddTypeAnnotation("months");
+            intValue.AddTypeAnnotation("days");
+            intValue.AddTypeAnnotation("hours");
+            intValue.AddTypeAnnotation("minutes");
+            intValue.AddTypeAnnotation("seconds");
+            var value = new IonStruct {{ "withannot", intValue }};
+            var reader = new UserTreeReader(value);
+
+            ReaderTestCommon.ReadAnnotations_SingleField(reader);
+        }
+
+        [TestMethod]
+        public void BlobTest()
+        {
+            //Must be in a struct:
+            // { blobbbb: {{data}} }
+            var arrayOfbytes = Enumerable.Repeat<byte>(1, 100).ToArray();         
+            ReadOnlySpan<byte> bytes = new ReadOnlySpan<byte>(arrayOfbytes);
+            var blob = _ionValueFactory.NewBlob(bytes);
+            var value = new IonStruct {{ "blobbbb", blob }};
+            var reader = new UserTreeReader(value);
+
+            ReaderTestCommon.Struct_OneBlob(reader);
+        }
+
+        [TestMethod]
+        public void BlobPartialReadTest()
+        {
+            var blob = new byte[30];
+            for (var i = 0; i < 30; i++)
+            {
+                blob[i] = (byte)i;
+            }
+            var value = _ionValueFactory.NewBlob(blob);
+            var reader = new UserTreeReader(value);
+
+            ReaderTestCommon.Blob_PartialRead(30, 7, reader);
+        }
+
+        [TestMethod]
+        public void SymbolTableTest()
```

Comment:
This should test that `UserTreeReader.GetSymbolTable` returns the correct symbol table. To test this, create a binary Ion stream that contains two different symbol tables, each with one user symbol value in scope. Make sure that the values have the correct text and that calling `UserTreeReader.GetSymbolTable` when positioned at each value returns the expected symbol table.

However, looking at the source code, I don't think this will work as-is because the UserTreeReader changes its current symbol table based on the IIonValues it retrieves from SystemTreeReader. As we know, SystemTreeReader does not surface IIonValue representations of symbol tables (see #50 ). As a result, UserTreeReader will never change its current symbol table.

Therefore, let's drop this test and make `UserTreeReader.GetSymbolTable` throw an exception until #50 is resolved.